File: public/snippets/Java/Switch Statement.java

/* -------------------------------------------------- */
/*      SYNTAX SWITCH STATEMENT PADA BAHASA JAVA      */
/* -------------------------------------------------- */
/*  switch(expression) {                              */
/*                                                    */
/*     case constant-expression  :                    */
/*        statement(s);                               */
/*        break; // optional                          */ 
/*                                                    */
/*     case constant-expression  :                    */
/*        statement(s);                               */
/*        break; // optional                          */
/*                                                    */
/*     default : // Optional                          */
/*     statement(s);                                  */
/*  }                                                 */
/* -------------------------------------------------- */

public class /* your file name */ {

   public static void main(String args[]) {
      
      char grade = 'C';

      switch(grade) {
         case 'A' :
            System.out.println("Excellent!"); 
            break;
         case 'B' :
         case 'C' :
            System.out.println("Well done");
            break;
         case 'D' :
            System.out.println("You passed");
         case 'F' :
            System.out.println("Better try again");
            break;
         default :
            System.out.println("Invalid grade");
      }
      System.out.println("Your grade is " + grade);
   }
}

File: public/snippets/C/Switch Statement.c
#include <stdio.h>

/*	--------------------------------------------------	*/
/*		  SYNTAX SWITCH STATEMENT PADA BAHASA C 		   */
/*	--------------------------------------------------	*/
/*  switch(expression) {								      */
/*														            */
/*     case constant-expression  : 						   */
/*        statement(s); 								      */
/*        break; // optional 							      */ 
/*														            */
/*     case constant-expression  :						   */
/*        statement(s);									      */
/*        break; // optional 							      */
/*  													            */
/*     default : // Optional 							      */
/*     statement(s);									         */
/*  }													            */
/*	--------------------------------------------------	*/
 
int main () {

   /* Deklarasi variabel lokal */
   char grade = 'B';

   switch(grade) {
      case 'A' :
         printf("Excellent!\n" );
         break;
      case 'B' :
      case 'C' :
         printf("Well done\n" );
         break;
      case 'D' :
         printf("You passed\n" );
         break;
      case 'F' :
         printf("Better try again\n" );
         break;
      default :
         printf("Invalid grade\n" );
   }
   
   printf("Your grade is  %c\n", grade );
 
   return 0;
}